src/OrchardCore/OrchardCore.Abstractions/Shell/Builders/ShellContextExtensions.cs

```diff
@@ -32,9 +30,10 @@ public static Task<(ILocker locker, bool locked)> TryAcquireShellActivateLockAsy
                 var options = shellContext.ServiceProvider.GetRequiredService<IOptions<ShellContextOptions>>();
                 timeout = TimeSpan.FromMilliseconds(options.Value.ShellActivateLockTimeout);
                 expiration = TimeSpan.FromMilliseconds(options.Value.ShellActivateLockExpiration);
+                lockName = options.Value.ShellActivateLockName;
```

Comment:
Why would it be different than the tenant name?

Reply:
Unless you want to block all tenants creation "by default"? But with this change this is what it would do, you couldn't create two tenants concurrently.

Reply:
I thought about this, but we may run many single-tenant instances on the same Reids in such case lockName always will be equal to "Default", the same issue with DPAPI Redis as I know we have a hardcoded "Default" tenant name if we run orchard in Single Tenant mode.

Reply:
For infos we already setup tenants concurently in unit tests but they all have different generated names

For info our RedisLock automaticaly prefix the lock key with the tenant name, so the unprefixed lock key should be the same, not an option that may be different for 2 instances running for the same tenant, unless we are sure they use the same config source, but i think it is better to let it hardcoded.

Hmm, but do you mean that you may run multiple single-tenant instances that are unrelated, not related to the same 
default tenant, but different ones using different databases ? But using the same Redis instance ?

Reply:
But i don't think so as your issue is a concurrent acces on the shell descriptor so in the same database and for the same tenant `Invalid serial number for shell descriptor`

Reply:
>but different ones using different databases ? But using the same Redis instance ?

Correct, we may run single-instance for each customer but share the same Redis instance, the configuration may vary depending on env. Testing Env, Staging, and Prod, but basically yes this is a scenario.
Having a Single Tenant gives us some benefits like updating the env without worrying about breaking other customers and so on.

Reply:
Okay so that's why you want an option e.g. to differentiate 2 different tenants, not related to the same database, but with the same name e.g. `Default`. Hmm, but then you will need the same e.g. if you want to share data with our RedisCache.

Through the redis config string we can define a `Default database index`, i think this is the right way to differentiate tenants not related to the same database, in place of changing / prefixing lock keys.

https://stackexchange.github.io/StackExchange.Redis/Configuration.html

Reply:
I may be wrong, but I think using DatabaseIndex as a Tenant delimiter is not a good idea, we might face some performance issues in such a case.